### PR TITLE
phylum-cli: adhere to formula standards

### PR DESCRIPTION
**Pull request by maxrake**
This change ensures the Phylum CLI is not built with the ability to
upgrade or uninstall itself, leaving those functions to `brew` instead.
It also generates and installs shell completion files. A caveat was
added to inform users that official extensions were not preinstalled.
Finally, the version command used for testing was altered to the form
that provides more succinct output so as not to overmatch on the assert.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/p/phylum-cli.rb [hunk 1]
```diff
@@ -21,12 +21,33 @@ class PhylumCli < Formula
 
   uses_from_macos "zlib"
 
+  # The `--no-default-features` option is used for Cargo commands to ensure the
+  # Phylum CLI default feature "selfmanage" is not activated. That feature allows
+  # for updating and uninstalling the CLI, which is undesired in this context:
+  # https://docs.brew.sh/Acceptable-Formulae#we-dont-like-tools-that-upgrade-themselves
```
Comment (chenrui333) [inline Formula/p/phylum-cli.rb:27]:
```suggestion
```

File: Formula/p/phylum-cli.rb [hunk 2]
```diff
@@ -21,12 +21,33 @@ class PhylumCli < Formula
...
   def install
-    system "cargo", "install", *std_cargo_args(path: "cli")
+    system "cargo", "install", "--no-default-features", *std_cargo_args(path: "cli")
+
+    # Generate and install shell completions
+    system "cargo", "run", "--package", "xtask", "--no-default-features", "gencomp"
+    bash_completion.install "target/completions/phylum.bash"
```
Comment (chenrui333) [inline Formula/p/phylum-cli.rb:33]:
```suggestion
    bash_completion.install "target/completions/phylum.bash" => "phylum"
```

File: Formula/p/phylum-cli.rb [hunk 3]
```diff
@@ -21,12 +21,33 @@ class PhylumCli < Formula
...
+  # Default official extensions are not installed because the standard install
+  # script asks the user for permission to overwrite existing extensions and
+  # blocking on user input is not desired in the context of a package manager.
+  def caveats
+    <<~EOS
+      No official extensions have been installed.
+      For a list of official extensions with installation instructions, see:
+        https://github.com/phylum-dev/cli/tree/main/extensions
+    EOS
   end
```
Comment (chenrui333) [inline Formula/p/phylum-cli.rb:47]: I feel like this can be removed as well as it is not homebrew specific

Comment (SMillerDev) [reply Formula/p/phylum-cli.rb:47]: For the homebrew specific part we can reduce it to:
```suggestion
  def caveats
    <<~EOS
      No official extensions have been preinstalled.
    EOS
  end
```

**Review by SMillerDev — APPROVED** [review]
Thanks @maxrake   ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request merged**